src/com/xilinx/rapidwright/design/ModuleInst.java

```diff
@@ -65,7 +65,10 @@ public class ModuleInst extends AbstractModuleInst<Module, Site, ModuleInst>{
     private ArrayList<Net> nets;
     /** A list of all NOCClients belonging to this instance **/
     private ArrayList<NOCClient> nocClients;
-
+    /** Keeps track of currently placed GND PIPs */
+    private Set<PIP> gndPIPs;
+    /** Keeps track of currently placed VCC PIPs */
+    private Set<PIP> vccPIPs;
     /**
```

Comment:
Does USED_NET pips need to be saved too?

Reply:
Good thought, but it never contains any PIPs.  Usually it is limited to site routing usage.